### PR TITLE
feat(native-app): Notifications V2

### PR DESCRIPTION
**Pull request by snaerseljan**
# Notifications V2

## What

This removes part from notification logic in the app. We are moving to pulling notifications from our api instead.
Notification are now pulled from our api, and the user is able to do infinite scrolling through the notifications.
Notifications have also received UI update.

Note! Expo-notifications handlers are broken in production. We will do a follow up pull request trying to fix it those event handlers.

## Screenshots / Gifs
Screenshot 2024-05-15 at 16 59 45

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced GraphQL queries and mutations for user notifications.
  - Added notification metadata fragments for improved notification data handling.

- **Improvements**
  - Enhanced notification handling logic for Android devices.
  - Improved error messages for document and notification loading.

- **Bug Fixes**
  - Fixed potential issue with incorrect license display during data fetching.

- **Refactor**
  - Replaced `apolloMKKVStorage` with `clearAllStorages` for better storage management.
  - Updated various components and hooks for improved performance and readability.

### CODE REVIEW

**Review by disaerna — APPROVED** [review]

File: apps/native/app/src/screens/notifications/notifications.tsx [hunk 1]
```diff
@@ -54,96 +52,123 @@ export const NotificationsScreen: NavigationFunctionComponent = ({
...
+        testID={testIDs.NOTIFICATION_CARD_BUTTON}
+      />
+    )
   }
 
-  const keyExtractor = useCallback((item: ListItem) => {
-    return item.id.toString()
-  }, [])
+  const keyExtractor = useCallback(
+    (item: ListItem, index: number) => item.id.toString(),
```
Comment (thoreyjona) [inline apps/native/app/src/screens/notifications/notifications.tsx:186]: ~Do we need the index here?

File: apps/native/app/src/ui/lib/card/notification-card.tsx [hunk 2]
```diff
@@ -81,13 +74,19 @@ export function NotificationCard({
...
+  testID,
 }: CardProps) {
   const theme = useTheme()
+
   return (
     <Host
       onPress={() => onPress(id)}
-      underlayColor={theme.isDark ? theme.shade.shade200 : '#EBEBFA'}
+      underlayColor={
+        underlayColor ?? theme.isDark ? theme.shade.shade200 : '#EBEBFA'
```
Comment (thoreyjona) [inline apps/native/app/src/ui/lib/card/notification-card.tsx:88]: What is this light color that does not have a color variable and why do we need it here?

**Review by thoreyjona — APPROVED** [review]
Nice 😄 
Only two minor comments 👌 
Did not test though.

File: apps/native/app/src/screens/notifications/notifications.tsx [hunk 3]
```diff
@@ -1,73 +1,192 @@
...
+  const [loadingMore, setLoadingMore] = useState(false)
+  const updateNavigationUnseenCount = useNotificationsStore(
+    ({ updateNavigationUnseenCount }) => updateNavigationUnseenCount,
+  )
 
-  const onNotificationPress = useCallback((notification: Notification) => {
-    navigateToNotification(notification, componentId)
+  const { data, loading, error, fetchMore, updateQuery } =
+    useGetUserNotificationsQuery({
+      fetchPolicy: 'network-only',
```
Comment (thoreyjona) [inline apps/native/app/src/screens/notifications/notifications.tsx:68]: Is this initial? Do we not want to fetch notifications from the cache?

**Review by saevarma — APPROVED** [review]
Core file LGTM

**Pull request merged**